### ISSUE TITLE
Veeam for Red Hat Hypervisor

### ISSUE THREAD

**Issue by RagingAdmin** [apache/cloudstack#6618]
Since Red Hat is now compatible with Veeam. Is there a plugin available? 

Veeam API is doing all the work, does the plugin need to be with vmware only. Other hyervisors can be added.

**Comment by rohityadavcloud** [comment]
Yes, subject to feasibility support for the RedHat/EL/KVM can be @RagingAdmin. Marking your issue as a feature request.

**Comment by NuxRo** [comment]
@RagingAdmin @rohityadavcloud the Veeam plugin in question is for RedHat Virtualization aka RHEV (oVirt basically). While this is indeed based on KVM, the plugin is very specific and will not work with anything else, ie generic KVM/libvirt is not supported.

**Comment by gitsan13** [comment]
Hi, I would like to contribute to this issue. I am currently undergoing this course of cloud computing and working on this issue will clearly enhance my learning. Can you please guide me through the steps?

**Comment by shwstppr** [comment]
@RagingAdmin cc @DaanHoogland @weizhouapache quite late but #12991 will be adding Veeam support for CloudStack KVM. We are in the final stages of testing. Though there is dependency on Veeam for adding a CloudStack KVM hypervisor manager plugin, it can be tested with the current oVirt plugin in Veeam.
Currently, the integrations enable Veeam-driven B&R with self-service backups coming in the future version when B&R APIs are made available for UHAPI platforms
More info here, https://github.com/apache/cloudstack/discussions/12869

**Comment by RagingAdmin** [comment]
Please note that Veeam also supports XCP-NG

https://community.veeam.com/blogs-and-podcasts-57/xcp-ng-plugin-for-veeam-backup-replication-public-beta-review-11677
________________________________

**Comment by yadvr** [comment]
@RagingAdmin Veeam indeed supports XCP-ng directly but not via CloudStack; B&R support in CloudStack requires the backups to be via the platform, otherwise operations such as restores or creating new VMs from backups would not reconcile cleanly to the platform (ACS) and it would not know about such changes done out of band to the resources it is tracking.
